### PR TITLE
Logrotate Cookbook 2.0

### PR DESCRIPTION
**Pull request by stevendanna**
- Convert the logrotate_app definition to a resource
- Improve platform support via attributes
- Further update the test infrastructure
- Replace Chefspec matcher

### CODE REVIEW

**Comment by johnbellone** [comment]
This looks good to me.

**Comment by m2rt** [comment]
When could this be merged? Its so great work that I will delay implementing logrotate cookbook before 2.0 is out. Really, good job.

Also, have you thought about using the ohai packages plugin and filtering the configuration options based on the version of packages installed?

**Comment by stevendanna** [comment]
@m2rt Going to do a bit more work on tit this weekend and merge it in the upcoming week

**Comment by stevendanna** [comment]
@m2rt Just wanted to follow up and let you know I haven't forgotten about this, just got bogged down in some other work.

File: resources/app.rb [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+#
+
+resource_name :logrotate_app
+
+property :path, [String, Array], required: true
+property :frequency, String, default: "weekly"
+property :template_mode, default: "0644"
+property :template_owner, default: "root"
+property :template_group, default: "root"
+property :base_dir, String, default: "/etc/logrotate.d"
```
Comment (ranjib) [inline resources/app.rb]: @stevendanna cookbook property ?

**Pull request merged**